lib/msf/core/rpc/v10/rpc_job_status_tracker.rb

```diff
@@ -0,0 +1,109 @@
+require 'monitor'
+
+class RpcJobStatusTracker
```

Comment:
This shouldn't be a global class, and since it doesn't do anything RPC specific, I would consider adding it `msf/base/simple` as well.

```suggestion
class Msf::Rpc::RpcJobStatusTracker
```